demos/common/models/src/detection_model_retinaface.cpp

```diff
@@ -0,0 +1,377 @@
+    int anchorNum, std::vector<ModelRetinaFace::Anchor>& anchors) {
+    auto desc = rawData->getTensorDesc();
+    auto sz = desc.getDims();
+    auto bboxPredLen = sz[1] / anchorNum;
+    auto blockWidth = sz[2] * sz[3];
+    LockedMemory<const void> outputMapped = rawData->rmap();
+    const float *memPtr = outputMapped.as<float*>();
+
+    for (auto i : indices) {
+        auto offset = blockWidth * bboxPredLen * (i % anchorNum) + (i / anchorNum);
+
+        auto dx = memPtr[offset];
+        auto dy = memPtr[offset + blockWidth];
+        auto dw = memPtr[offset + blockWidth * 2];
+        auto dh = memPtr[offset + blockWidth * 3];
+
+        auto predCtrX = dx * anchors[i].getWidth() + anchors[i].getXCenter();
+        auto predCtrY = dy * anchors[i].getHeight() + anchors[i].getYCenter();
+        auto predW = exp(dw) * anchors[i].getWidth();
+        auto predH = exp(dh) * anchors[i].getHeight();
+
+        bboxes->push_back({ predCtrX - 0.5 * (predW - 1.0), predCtrY - 0.5 * (predH - 1.0),
+            predCtrX + 0.5 * (predW - 1.0), predCtrY + 0.5 * (predH - 1.0) });
+    }
+}
+
+
+void filterLandmarks(std::vector<cv::Point2f>* landmarks, const std::vector<size_t>& indices, InferenceEngine::MemoryBlob::Ptr rawData,
+    int anchorNum, const std::vector<ModelRetinaFace::Anchor>& anchors, double landmarkStd) {
+    auto desc = rawData->getTensorDesc();
+    auto sz = desc.getDims();
+    auto landmarkPredLen = sz[1] / anchorNum;
+    auto blockWidth = sz[2] * sz[3];
+    LockedMemory<const void> outputMapped = rawData->rmap();
+    const float *memPtr = outputMapped.as<float*>();
+
+    for (auto i : indices) {
+        auto ctrX = anchors[i].getXCenter();
+        auto ctrY = anchors[i].getYCenter();
+        for (int j = 0; j < ModelRetinaFace::LANDMARKS_NUM; ++j) {
+            auto offset = (i % anchorNum) * landmarkPredLen * sz[2] * sz[3] + i / anchorNum;
+            auto deltaX = memPtr[offset + j * 2 * blockWidth] * landmarkStd;
+            auto deltaY = memPtr[offset + (j * 2 + 1)*blockWidth] * landmarkStd;
+            landmarks->push_back({ static_cast<float>(deltaX * anchors[i].getWidth() + anchors[i].getXCenter()),
+                static_cast<float>(deltaY * anchors[i].getHeight() + anchors[i].getYCenter()) });
+        }
+    }
+}
+
+void filterMasksScores(std::vector<double>* masks, const std::vector<size_t>& indices, InferenceEngine::MemoryBlob::Ptr rawData, int anchorNum) {
+    auto desc = rawData->getTensorDesc();
+    auto sz = desc.getDims();
+    auto start = sz[2] * sz[3] * anchorNum * 2;
+    LockedMemory<const void> outputMapped = rawData->rmap();
+    const float *memPtr = outputMapped.as<float*>();
+
+    for (auto i : indices) {
+        auto offset = (i % anchorNum) * sz[2] * sz[3] + i / anchorNum;
+        masks->push_back(memPtr[start + offset]);
+    }
+}
+
+std::vector<int> nms(const std::vector<ModelRetinaFace::Anchor>& boxes, const std::vector<double>& scores, double thresh) {
```

Comment:
do you not have already implemented NMS code for detections, I believe it is a common postprocessing step (e.g. as I know also at least used for yolo)?

Reply:
No, we don't have common NMS function in this demo:(